src/app/button-page/button-page.component.html

```diff
@@ -1,6 +1,13 @@
 <div class="board" *ngIf="board; else noBoard">
-  <div class="boardRow" [style.height]="calculateRowHeight()" *ngFor="let row of board.grid.order">
-    <div class="boardButton" *ngFor="let cell of row">
+  <div class="boardRow"
+      [class.scanHighlight]="currentScanHighlightRow(row)"
+      [class.scanSelect]="currentScanSelectRow(row)"
```

Comment:
You could have a model for this component which includes highlight and selection state. The model could combine your other two models.  Then you wouldn't have to pass row into methods to determine these states.